NHibernate.Spatial.PostGis/Dialect/PostGisDialect.cs

```diff
@@ -270,12 +270,17 @@ public virtual SqlString GetSpatialAggregateString(object geometry, SpatialAggre
                 default:
                     throw new ArgumentException("Invalid spatial aggregate argument");
             }
-            return new SqlStringBuilder()
-                .Add(aggregateFunction)
+            var builder = new SqlStringBuilder();
+            builder.Add(aggregateFunction)
                 .Add("(")
                 .AddObject(geometry)
-                .Add(")")
-                .ToSqlString();
+                .Add(")");
+            // Convert to geometry as there is no binary output function available for type box2d type
+            if (aggregate == SpatialAggregate.Envelope)
+            {
+                builder.Add("::geometry");

```

Comment:
Used the same approach as [Hibernate Spatial did](https://github.com/hibernate/hibernate-orm/blob/a7fc440e4c533a32b956044c9c421727d70afa79/hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisFunctions.java#L200). This is a limitation of `PostGIS` as does not have a binary representation of `box2d` type. By casting it to `geometry` a `polygon` is returned.